internal/test/daemon/plugin.go

```diff
@@ -66,12 +67,9 @@ func withPluginInspect(name string, f func(*types.Plugin, poll.LogT) poll.Result
 
 }
 
-func withClient(d *Daemon, f func(client.APIClient, poll.LogT) poll.Result) func(poll.LogT) poll.Result {
-	return func(t poll.LogT) poll.Result {
-		c, err := d.NewClient()
-		if err != nil {
-			poll.Error(err)
-		}
-		return f(c, t)
+func withClient(t assert.TestingT, d *Daemon, f func(client.APIClient, poll.LogT) poll.Result) func(poll.LogT) poll.Result {
+	return func(pt poll.LogT) poll.Result {
+		c := d.NewClientT(t)
```

Comment:
Wondering if there's a need to use `poll.Error` here? In that case perhaps we should inline ~ what `NewClientT` does here

```suggestion
		if ht, ok := t.(test.HelperT); ok {
			ht.Helper()
		}
		c, err := client.NewClientWithOpts(
			client.FromEnv,
			client.WithHost(d.Sock()))

		if err != nil {
			poll.Error(err)
		}
```

@vdemeester ?

Reply:
`NewClientT` will mark the build as fail (and fatal), so I'm pretty sure it will get out of the poll loop in case of failure… *but* I'm not 100% sure about that.
cc @dnephin

Reply:
Ya, that sounds right. `t.Fatal` will panic and stop execution.

Reply:
Thanks @dnephin - looks like ready to merge then 👍